### PR TITLE
Mods to schroedinger

### PR DESCRIPTION
**Pull request by raynol-dsouza**
I added the conversion factor to solve the TISE in pyiron units. The resulting curves look exactly like how I have them in my notebook, However, having the mass units in AMU makes the demo notebooks give some not-so-nice looking plots. I will modify the notebooks/class accordingly after I receive Liam's input.

### CODE REVIEW

**Comment by liamhuber** [comment]
@raynol-dsouza do you understand why all the elasticity stuff is showing up in the diff? I just merged master into schroedinger and thought that would get rid of it, but it's still there.

File: pyiron_continuum/schroedinger/schroedinger.py [hunk 1]
```diff
@@ -101,7 +104,8 @@ def get_boltzmann_occupation(self, temperature):
     def get_boltzmann_rho(self, temperature):
         if self.psi is not None:
             w = self.get_boltzmann_occupation(temperature)
-            return self._weight_states(self.rho, w).sum(axis=0)
+            weighted_rho = self._weight_states(self.rho, w).sum(axis=0)
+            return weighted_rho / weighted_rho.sum()
```
Comment (liamhuber) [inline pyiron_continuum/schroedinger/schroedinger.py:108]: can you double check that this normalization is necessary? IIRC the boltzmann weights are now normalized by their sum (i.e. w/Z) so this should already be normalized.

**Comment by raynol-dsouza** [comment]
I have no idea why @samwaseda's commits are showing up here.

File: pyiron_continuum/schroedinger/schroedinger.py (hunk 1 above)
Comment (raynol-dsouza) [reply pyiron_continuum/schroedinger/schroedinger.py:108]: You are right. It is already normalized. I will revert this part.

File: pyiron_continuum/schroedinger/schroedinger.py [hunk 2]
```diff
@@ -26,9 +26,12 @@
 
 
 # TODO: Convert to pyiron units
-HBAR = 1  # EV_TO_U_ANGSTROMSQ_PER_SSQ * physical_constants['Planck constant in eV/Hz'][0] / (2 * np.pi)
-M_EL = 1  # physical_constants['electron mass in u'][0]
+HBAR = physical_constants['reduced Planck constant in eV s'][0]
 KB = physical_constants['Boltzmann constant in eV/K'][0]
+# conversion factor to convert the units of all terms in the Schroedinder equation are in eV. More documentation
+# in _hamiltonian of the TISE class.
+EV_S_PER_ANG_SQUARED_PER_AMU_IN_EV = 9.64853322e27
```
Comment (liamhuber) [inline pyiron_continuum/schroedinger/schroedinger.py:33]: Name is wrong, it's (eV s)^2. Writing 'squared' in all places will be too long though, even for me.

What about `EV2_S2_PER_ANG2_AMU2_IN_EV`?

File: pyiron_continuum/schroedinger/schroedinger.py [hunk 3]
```diff
@@ -157,12 +161,11 @@ def _hamiltonian(self, psi):
...
 
-        Thus, to get the first term to eV we need hbar in units of sqrt(eV Angstroms^2 u).
-        Starting with hbar in eVs, we need to convert eV to (Angstroms^2 u / s^2) s.t. the first term comes out with eV.
-        https://www.wolframalpha.com/input/?i=1+%28atomic+mass+units%29*%28angstroms%5E2%29%2F%28s%5E2%29+in+eV
-        Except that didn't seem to work, so for now just use HBAR as defined in the header
+        Thus, to get the first term to eV we need to multiply it by eV s / A**2 / u.
+        The math is here: https://www.wolframalpha.com/input/?i=%28eV*s%29%5E2%2F%28AMU%29+%2F%28Angstrom%5E2%29+in+eV
         """
-        return -(HBAR**2 / (2 * self.input.mass)) * self.mesh.laplacian(psi) + self._potential_psi(psi)
+        return -(HBAR ** 2 / (2 * self.input.mass) * EV_S_PER_ANG_SQUARED_PER_AMU_IN_EV) * self.mesh.laplacian(psi) + \
```
Comment (liamhuber) [inline pyiron_continuum/schroedinger/schroedinger.py:167]:
```suggestion
        return -(HBAR ** 2 / (2 * self.input.mass)) * self.mesh.laplacian(psi) * EV_S_PER_ANG_SQUARED_PER_AMU_IN_EV + \
```

This is a deep nit. I just don't like the unit conversion factor breaking up all the things that have units (the laplacian is what gives us our per ang^2)

File: pyiron_continuum/schroedinger/schroedinger.py (hunk 2 above)
Comment (raynol-dsouza) [reply pyiron_continuum/schroedinger/schroedinger.py:33]: It's supposed to be (eV s / A)^2 / AMU. 

> What about EV2_S2_PER_ANG2_AMU2_IN_EV

Reads okay.

Comment (liamhuber) [reply pyiron_continuum/schroedinger/schroedinger.py]: ah, yeah, `AMU` not `AMU2`

**Comment by raynol-dsouza** [comment]
Presently it is 1 AMU. We could discuss it during the next pyiron meeting and modify it then.

**Pull request merged**